crates/matrix-sdk/src/send_queue.rs

```diff
@@ -675,13 +688,21 @@ impl QueueStorage {
     async fn mark_as_sent(
         &self,
         transaction_id: &TransactionId,
+        event_id: &EventId,
     ) -> Result<(), RoomSendQueueStorageError> {
         // Keep the lock until we're done touching the storage.
         let mut being_sent = self.being_sent.write().await;
         being_sent.remove(transaction_id);
 
-        let removed =
-            self.client()?.store().remove_send_queue_event(&self.room_id, transaction_id).await?;
+        let client = self.client()?;
+        let store = client.store();
+
+        // Update all dependent events.
```

Comment:
We're holding onto the `being_sent` lock which acts as a lock for storage too, so there's no way to abort in the middle, as far as I can tell.

Reply:
Err, the lock prevents concurrent access, it does not prevent the client suddenly closing or the device losing power. What I'm asking is, how bad will the inconsistency be if things get aborted in the middle here.

Reply:
Since the parent event is deleted after the dependent events, worst case is we retry to send it (which is fine, because it'll reuse the same txn id), and re process pending edits/redacts thereafter.